homeassistant/components/rainforest_raven/sensor.py

```diff
@@ -0,0 +1,193 @@
+"""Sensor entity for a Rainforest RAVEn device."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+    StateType,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_MAC,
+    PERCENTAGE,
+    EntityCategory,
+    UnitOfEnergy,
+    UnitOfPower,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import RAVEnDataCoordinator
+
+
+@dataclass
```

Comment:
```suggestion
@dataclass(frozen=True)
```

Reply:
That doesn't seem to work:
```
error: Cannot inherit frozen dataclass from a non-frozen one
```

Reply:
It should work now after you merged in dev to your PR branch, please give it another try.

Reply:
Ah, I understand. Thanks, it's working now. Implemented in 7317d54.